doordeck-sdk/src/commonMain/kotlin/com/doordeck/multiplatform/sdk/internal/api/AccountClient.kt

```diff
@@ -79,10 +106,23 @@ internal open class AccountClient(
      * @see <a href="https://developer.doordeck.com/docs/#verify-ephemeral-key-registration">API Doc</a>
      */
     suspend fun verifyEphemeralKeyRegistrationRequest(code: String, privateKey: ByteArray): RegisterEphemeralKeyResponse {
+        return verifyEphemeralKeyRegistration(code, privateKey)
+    }
+
+    suspend fun verifyEphemeralKeyRegistrationWithContextRequest(code: String): RegisterEphemeralKeyResponse {
+        val privateKey = contextManager.getPrivateKey()
+            ?: throw MissingContextFieldException("Key pair is missing")
+        return verifyEphemeralKeyRegistration(code, privateKey)
+    }
+
+    private suspend fun verifyEphemeralKeyRegistration(code: String, privateKey: ByteArray): RegisterEphemeralKeyResponse {
         val codeSignature = code.signWithPrivateKey(privateKey).encodeByteArrayToBase64()
-        return httpClient.post(Paths.getVerifyEphemeralKeyRegistrationPath()) {
+        return httpClient.post<RegisterEphemeralKeyResponse>(Paths.getVerifyEphemeralKeyRegistrationPath()) {
             addRequestHeaders()
             setBody(VerifyEphemeralKeyRegistrationRequest(codeSignature))
+        }.also {
+            contextManager.setUserId(it.userId)
+            contextManager.setCertificateChain(it.certificateChain)
```

Comment:
From now on, we will update the context values whenever possible. Previously, this had to be done manually by the developer. Please note that this only updates the variable; it does not persist it.